.github/workflows/build-test-e2e.yml

```diff
@@ -0,0 +1,52 @@
+name: 'Build/Run End-to-End Tests'
+
+on: push
+jobs:
+  build-and-run-e2e:
+    if: ${{ github.ref == 'develop' || github.ref == 'release-**' || github.actor == github.repository_owner }}
+    runs-on: macos-latest
```

Comment:
Tests failed intermittently in CI with GitHub standard Linux runner. Using the standard macOS runner, which is larger, eliminated flake in CI.

Reply:
If we decide to use a GitHub-hosted larger runner, tweak this so that a standard runner is still used when devs run the workflow in their forked repo. Otherwise, they will be charged for using the larger runner in their personal account.